.github/workflows/main.yml

```diff
@@ -83,12 +83,6 @@ jobs:
       run: |
         pip install --use-feature=2020-resolver --progress-bar off .[test]
 
-    - name: Install Forte-wrappers-spacy
```

Comment:
"use the workflow matrix set to run different dependencies as different runs."
Do you mean you will have different versions of forte-wrapper and at that time you want to define a matrix testing different versions of them?

Reply:
"use the workflow matrix set to run different dependencies as different runs."
---------------
Do you mean in the future we will have different versions of forte-wrapper, and you want to define a matrix to test them all?

Reply:
1. No, not different versions of `forte-wrapper`. I show you the `forte-wrapper` repository workflow as an example of how to set up workflow matrix.
2. In this repo, ForteHealth, we can setup the same matrix, since not all processor in this repo can be installed and run at the same time, they can be tested individually. Note that you could use the `extra_install` method in `setup.py` to specify different sets of dependencies.

You may wonder how we make them work together in the future, that's another issue (we have solutions such as RemoteProcessor or Docker images). Right now we can focus on making them each testable.

Reply:
I still have a question: @Piyush13y seems to want to set forte.spacy as an extra_install. If so, forte.spacy and neurocoref will be in the same extra_install list, example:
```
setuptools.setup(
....
    install_requires=[
        "forte~=0.2.0",
    ],
    extras_require={
        .....
        "coreference": [
            "forte.spacy",  # TODO: version
            "cython>=0.25",
            "neuralcoref @ git+https://git@github.com/huggingface/neuralcoref.git@4.0.0#egg=neuralcoref",
        ],
    },
    include_package_data=True,
    python_requires=">=3.6",
)
```

I think pip won't make sure that forte.spacy is install ahead of neuralcoref.

Reply:
> @Piyush13y seems to want to set forte.spacy as an extra_install.

Emmm... It's a little bit confusing because now forte.spacy is not even in the setup.py in master branch: https://github.com/asyml/ForteHealth/blob/master/setup.py

Reply:
1. If the `corference` processor require `spacy` and then yes it should be included in the `coreference` extra install section. You just need to specify the versions that work. We can have other spacy versions in other sections
2. I think the master `setup.py` miss the necessary settings:
   - what happen right now is we do testing through the CI workflow and install spacy wrapper [there](https://github.com/asyml/ForteHealth/blob/master/.github/workflows/main.yml#L86). The tests can certainly run, but if some users really want to use this library, they have to manually install those dependencies. 
   - the correct solution is to specify the correct `extra_require` for each processor, and in the work flow, we trigger the `extra_install` via `pip`. And when user want to use some processors, they can also use `pip`

Reply:
OK. Actually what neuralcoref requires is spacy not forte.spacy, but I find that the auto-downloading function (`load_lang_model`) in forte.spacy is really useful, so that's why I chose forte.spacy not the raw spacy.

And *that* is why i have to build neuralcoref from source, since the spacy inside forte.spacy is newer than the one required by neuralcoref.

In this case, do you think what is the correct dependency to use, spacy or forte.spacy?

Reply:
Installing a spacy version compatible with neuralcoref causes the failure of `SpacyProcessor`. Since `SpacyProcessor` is needed in unit test, this seems to be a dead end before we can manage to run the modules separately.

Reply:
now we come back to my initial suggestion, use the `matrix` set up to run the module separately

Reply:
The latest release of neuralcoref is in 2019. Maybe it is worth considering changing a better package.

Reply:
huh, actually for now we are simply looking for a coreference module that works. Maybe we can directly add this function to [AllenNLPProcessor](https://github.com/asyml/forte-wrappers/blob/main/src/allennlp/fortex/allennlp/allennlp_processors.py), or other packages you find easy